UnitySDK/Assets/ML-Agents/Scripts/Agent.cs

```diff
@@ -531,21 +521,15 @@ void ResetData()
 
             BrainParameters param = brain.brainParameters;
             actionMasker = new ActionMasker(param);
-            // If we haven't initialized vectorActions, initialize to 0. This should only
-            // happen during the creation of the Agent. In subsequent episodes, vectorAction
-            // should stay the previous action before the Done(), so that it is properly recorded. 
-            if (action.vectorActions == null)
```

Comment:
Looks like this removes a check -- is that intentional?

Reply:
It is not. Fixing now.